x/mint/abci_test.go

```diff
@@ -0,0 +1,171 @@
+	yearOne := time.Date(2024, 1, 1, 1, 1, 1, 1, time.UTC).UTC()
+	yearTwo := time.Date(2025, 1, 1, 1, 1, 1, 1, time.UTC).UTC()
+	yearTwenty := time.Date(2043, 1, 1, 1, 1, 1, 1, time.UTC).UTC()
+
+	type testCase struct {
+		name string
+		ctx  sdk.Context
+		want sdk.Dec
+	}
+
+	testCases := []testCase{
+		{
+			name: "inflation rate is 0.08 initially",
+			ctx:  ctx.WithBlockHeight(0).WithBlockTime(unixEpoch),
+			want: sdk.NewDecWithPrec(8, 2), // 0.08
+		},
+		{
+			name: "inflation rate is 0.08 for year zero",
+			ctx:  ctx.WithBlockHeight(1).WithBlockTime(yearZero),
+			want: sdk.NewDecWithPrec(8, 2), // 0.08
+		},
+		{
+			name: "inflation rate is 0.08 for year zero and six months",
+			ctx:  ctx.WithBlockTime(yearZeroAndSixMonths),
+			want: sdk.NewDecWithPrec(8, 2), // 0.08
+		},
+		{
+			name: "inflation rate is 0.072 for year one",
+			ctx:  ctx.WithBlockTime(yearOne),
+			want: sdk.NewDecWithPrec(72, 3), // 0.072
+		},
+		{
+			name: "inflation rate is 0.0648 for year two",
+			ctx:  ctx.WithBlockTime(yearTwo),
+			want: sdk.NewDecWithPrec(648, 4), // 0.0648
+		},
+		{
+			name: "inflation rate is 0.015 for year twenty",
+			ctx:  ctx.WithBlockTime(yearTwenty),
+			want: sdk.NewDecWithPrec(15, 3), // 0.015
+		},
+	}
+
+	for _, tc := range testCases {
+		mint.BeginBlocker(tc.ctx, app.MintKeeper)
+		got, err := app.MintKeeper.InflationRate(ctx, &minttypes.QueryInflationRateRequest{})
+		assert.NoError(t, err)
+		assert.Equal(t, tc.want, got.InflationRate)
+	}
+}
+
+func TestAnnualProvisions(t *testing.T) {
+	app, _ := util.SetupTestAppWithGenesisValSet()
+	ctx := sdk.NewContext(app.CommitMultiStore(), types.Header{}, false, tmlog.NewNopLogger())
+	unixEpoch := time.Unix(0, 0).UTC()
+	yearZero := time.Date(2023, 1, 1, 1, 1, 1, 1, time.UTC).UTC()
+	yearZeroAndSixMonths := time.Date(2023, 6, 1, 1, 1, 1, 1, time.UTC).UTC()
+	yearOne := time.Date(2024, 1, 1, 1, 1, 1, 1, time.UTC).UTC()
+	yearTwo := time.Date(2025, 1, 1, 1, 1, 1, 1, time.UTC).UTC()
+	yearTwenty := time.Date(2043, 1, 1, 1, 1, 1, 1, time.UTC).UTC()
+
+	type testCase struct {
+		name string
+		ctx  sdk.Context
+		want sdk.Dec
+	}
+
+	testCases := []testCase{
+		{
+			name: "annual provisions is 80,000 initially",
+			ctx:  ctx.WithBlockHeight(0).WithBlockTime(unixEpoch),
+			want: sdk.NewDec(80_000), // 1,000,000 (total supply) * 0.08 (inflation rate)
+		},
+		{
+			name: "annual provisions is 80,000 for year zero",
+			ctx:  ctx.WithBlockHeight(1).WithBlockTime(yearZero),
+			want: sdk.NewDec(80_000), // 1,000,000 (total supply) * 0.08 (inflation rate)
+		},
+		{
+			name: "annual provisions is 80,000 for year zero and six months",
+			ctx:  ctx.WithBlockTime(yearZeroAndSixMonths),
+			want: sdk.NewDec(80_000), // 1,000,000 (total supply) * 0.08 (inflation rate)
+		},
+		{
+			name: "annual provisions is 72,000 for year one",
+			ctx:  ctx.WithBlockTime(yearOne),
+			want: sdk.NewDec(72_000), // 1,000,000 (total supply) * 0.072 (inflation rate)
```

Comment:
I'm guessing this doesn't take into consideration the compounding effects of inflation?